### PR TITLE
7. POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by NicollahSekete**
Request body accepts:

an object with the following properties:
username
body
Responds with:

the posted comment

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -111,4 +112,52 @@ describe("app", () => {
...
+                body: 'such a big fan wow'
+            }).expect(201).then((res) => {
+
+                const comment = res.body.comment
+
+                expect(comment.body).toBe("such a big fan wow");
+                expect(comment.author).toBe("icellusedkars");
+                expect(comment.article_id).toBe(2);
+                expect(comment.votes).toBe(0);
+
```
Comment (sirKevlar) [inline __tests__/app.test.js:188]: As we know the exact comment we are getting back here, we can assert against all of the individual properties (you have done this for half of them above). Then we no longer need the match object assertion directly below.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -111,4 +112,52 @@ describe("app", () => {
...
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
+                });
+            })
+        })
+
```
Comment (sirKevlar) [inline __tests__/app.test.js:199]: It is possible for our client to include some unnecessary properties on the post body. Here we could use an additional 201 - ignores unnecessary properties

File: __tests__/app.test.js [hunk 3]
```diff
@@ -111,4 +112,52 @@ describe("app", () => {
...
+        test("expect 400 when missing body", ()=>{
+            return request(app).post("/api/articles/2/comments").send({
+                username: 'icellusedkars',
+                body: ''
+            }).expect(400).then(({body}) => {
+                expect(body.msg).toBe('Bad Request')
+            })
+
+        })
+
```
Comment (sirKevlar) [inline __tests__/app.test.js:229]: We will need to do the 2x tests for bad parametric endpoint here please:
400: invalid id ("/api/articles/not-a-number/comments")
404: valid but non-existent id ("/api/articles/7777777/comments")

File: __tests__/app.test.js [hunk 4]
```diff
@@ -111,4 +112,52 @@ describe("app", () => {
...
+        
+    })
```
Comment (sirKevlar) [inline __tests__/app.test.js:251]: It is also possible that we could be missing the username or body properties altogether, we need a 'sad path' test for this.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -118,46 +118,78 @@ describe("app", () => {
...
+            return request(app).post("/api/articles/noNumber/comments").send({
+                username: 'icellusedkars',
+                body: 'heres the body'
+            }).expect(400).then(({ body }) => {
+                expect(body.msg).toBe('Bad Request')
+            })
+        })
+
+
+        test("expect 404 when missing body", () => {
```
Comment (HannahBurks) [inline __tests__/app.test.js]: We seem to have the wrong description on this one - 404 non existent ID is what we want for this one 😃

File: __tests__/app.test.js [hunk 6]
```diff
@@ -111,4 +112,84 @@ describe("app", () => {
 
     })
 
+    describe("POST /api/articles/:article_id/comments", () => {
```
Comment (HannahBurks) [inline __tests__/app.test.js:173]: 2 more tests we want to add here.
test for if the article ID is invalid - ie it is a string (400)
test for it the username is not valid (isn't in the users table) - should throw a PSQL error that can be dealt with - 404

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Hey Nicollah - well done, looking great, just needs a couple more tests adding/tweaking. Almost there with this one 😄

**Pull request merged**